sql/step_a115_add_missing_foreign_key_for_auditable_entities.sql

```diff
@@ -0,0 +1,95 @@
+ADD CONSTRAINT FK_MEDICALDSRINVENTORYROW_USER_1
+FOREIGN KEY (MINVTR_CREATED_BY) REFERENCES OH_USER (US_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+ALTER TABLE OH_MEDICALDSRINVENTORYROW 
+ADD CONSTRAINT FK_MEDICALDSRINVENTORYROW_USER_2
+FOREIGN KEY (MINVTR_LAST_MODIFIED_BY) REFERENCES OH_USER (US_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+ALTER TABLE OH_MEDICALDSRINVENTORYROW 
+DROP CONSTRAINT oh_medicaldsrinventoryrow_ibfk_1;
+
+ALTER TABLE OH_MEDICALDSRINVENTORYROW 
+ADD CONSTRAINT FK_MEDICALDSRINVENTORYROW_MEDICALDSRINVENTORY
+FOREIGN KEY (MINVTR_INVT_ID) REFERENCES OH_MEDICALDSRINVENTORY (MINVT_ID) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+ALTER TABLE OH_MEDICALDSRINVENTORYROW 
+DROP CONSTRAINT oh_medicaldsrinventoryrow_ibfk_2;
+
+ALTER TABLE OH_MEDICALDSRINVENTORYROW 
+ADD CONSTRAINT FK_MEDICALDSRINVENTORYROW_MEDICALDSR
+FOREIGN KEY (MINVTR_MDSR_ID) REFERENCES OH_MEDICALDSR (MDSR_ID) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+ALTER TABLE OH_MEDICALDSRINVENTORYROW 
+DROP CONSTRAINT oh_medicaldsrinventoryrow_ibfk_3;
+
+ALTER TABLE OH_MEDICALDSRINVENTORYROW 
+ADD CONSTRAINT FK_MEDICALDSRINVENTORYROW_MEDICALDSRLOT
+FOREIGN KEY (MINVTR_LT_ID_A) REFERENCES OH_MEDICALDSRLOT (LT_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+-- OH_MEDICALDSRSTOCK --
+
+ALTER TABLE OH_MEDICALDSRSTOCK
+MODIFY COLUMN MS_CREATED_BY VARCHAR(50) CHARACTER SET utf8 COLLATE utf8_general_ci;
+
+ALTER TABLE OH_MEDICALDSRSTOCK
+MODIFY COLUMN MS_LAST_MODIFIED_BY VARCHAR(50) CHARACTER SET utf8 COLLATE utf8_general_ci;
+
+ALTER TABLE OH_MEDICALDSRSTOCK
+ADD CONSTRAINT FK_MEDICALDSRSTOCK_USER_1
+FOREIGN KEY (MS_CREATED_BY) REFERENCES OH_USER (US_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+ALTER TABLE OH_MEDICALDSRSTOCK 
+ADD CONSTRAINT FK_MEDICALDSRSTOCK_USER_2
+FOREIGN KEY (MS_LAST_MODIFIED_BY) REFERENCES OH_USER (US_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+
+-- OH_USER_SETTINGS --
+
+ALTER TABLE OH_USER_SETTINGS
+ADD CONSTRAINT FK_USER_SETTINGS_USER_1
+FOREIGN KEY (USS_CREATED_BY) REFERENCES OH_USER (US_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+ALTER TABLE OH_USER_SETTINGS
+ADD CONSTRAINT FK_USER_SETTINGS_USER_2
+FOREIGN KEY (USS_LAST_MODIFIED_BY) REFERENCES OH_USER (US_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+-- OH_PATIENT_CONSENSUS --
+
+ALTER TABLE OH_PATIENT_CONSENSUS
+ADD CONSTRAINT FK_PATIENT_CONSENSUS_USER_1
+FOREIGN KEY (PTC_CREATED_BY) REFERENCES OH_USER (US_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+ALTER TABLE OH_PATIENT_CONSENSUS
+ADD CONSTRAINT FK_PATIENT_CONSENSUS_USER_2
+FOREIGN KEY (PTC_LAST_MODIFIED_BY) REFERENCES OH_USER (US_ID_A) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+ALTER TABLE OH_PATIENT_CONSENSUS 
+DROP CONSTRAINT oh_patient_consensus_ibfk_1;
+
+SET FOREIGN_KEY_CHECKS = 0;
+
+ALTER TABLE OH_PATIENT_CONSENSUS 
+ADD CONSTRAINT FK_PATIENT_CONSENSUS_PATIENT
+FOREIGN KEY (PTC_PAT_ID) REFERENCES OH_PATIENT (PAT_ID) ON DELETE NO ACTION ON UPDATE NO ACTION;
+
+SET FOREIGN_KEY_CHECKS = 1;
```

Comment:
I don't think they are needed:

```suggestion

ALTER TABLE OH_PATIENT_CONSENSUS 
ADD CONSTRAINT FK_PATIENT_CONSENSUS_PATIENT
FOREIGN KEY (PTC_PAT_ID) REFERENCES OH_PATIENT (PAT_ID) ON DELETE NO ACTION ON UPDATE NO ACTION;

```

Reply:
without those param I have this error: `ERROR 1452 (23000) at line 96 in file: 'step_a115_add_missing_foreign_key_for_auditable_entities.sql': Cannot add or update a child row: a foreign key constraint fails (`oh`.`#sql-25c8_416`, CONSTRAINT `FK_PATIENT_CONSENSUS_PATIENT` FOREIGN KEY (`PTC_PAT_ID`) REFERENCES `oh_patient` (`PAT_ID`) ON DELETE NO ACTION ON UPDATE NO ACTION)`

Reply:
looks demo data are wrong, with 48 patient consensus record referring to not existing PAT_ID, and the error was introduced with 2ed7be03a649c999a298a40cb4471eb3f072bedf.
You can safely remove those checks and I'm opening an issue to fix the demo data.
If you want you can first wait for it to be merged.